src/libraries/Microsoft.PowerFx.Core/Public/Types/RecordType.cs

```diff
@@ -85,6 +85,11 @@ public virtual RecordType Add(NamedFormulaType field)
         /// </summary>
         public RecordType Add(string logicalName, FormulaType type, string optionalDisplayName = null)
         {
+            if (type._type.IsDeferred)
```

Comment:
> if (type._type.IsDeferred) [](http://example.com/codeflow?start=11&length=27)

nit - this check should actually be down a layer in DType so that it can't be circumvented. 

FormulaType is just a lightweight wrapper

Reply:
ditto for checks in AggregateType / TableType